File: backend/src/main/java/backend/domain/dto/ProductDto.java
package backend.domain.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import backend.rdb.entities.enums.ProductCategory;
import backend.rdb.entities.enums.SaleStatus;


@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ProductDto {
    private String id;
    private String fullName;
    private Integer price;
    private String brand;
    private String model;
    private Double avgGrade;
    private SaleStatus saleStatus;
    private ProductCategory productCategory;
    private Boolean isAccess;
    private String description;
    private String imageUrl;
    private String videoUrl;
    private String color;
}

File: frontend/src/app/dto/ProductDto.ts
import {SafeUrl} from "@angular/platform-browser";

export interface ProductDto {
  id: string
  display: string | null;
  frequencies: string | null;
  camera: string | null;
  versionOS: string | null;
  batteryPower: string | null;
  otherFunctions: string | null;
  is3g: boolean | null;
  isLTE: boolean | null;
  is5g: boolean | null;
  isNanoSim: boolean | null;
  is_2Sim: boolean | null;
  productId: string | null
  fullName: string | null
  price: Number | null
  brand: string | null
  model: string | null
  avgGrade: Number | null
  saleStatus: string | null
  productCategory: string
  isAccess: boolean | null
  description: string | null
  color : string | null
  imageUrl: string | null | SafeUrl
  videoUrl: string | null
}
